src/main/java/com/couchbase/lite/replicator/Replication.java

```diff
@@ -406,20 +404,76 @@ public int getChangesCount() {
     /**
      * Update the lastError
      */
-    /* package */ void setLastError(Throwable lastError) {
+    protected void setLastError(Throwable lastError) {
         this.lastError = lastError;
     }
 
-    /* package */ String remoteCheckpointDocID() {
+    /**
+     * Following two methods for temporary methods instead of CBL_ReplicatorSettings implementation.
+     */
+    protected String remoteCheckpointDocID() {
         return replicationInternal.remoteCheckpointDocID();
     }
 
+    public Set<String> getPendingDocumentIDs(){
+        if(isPull() || (isRunning() && pendingDocIDs != null))
+            return pendingDocIDs;
```

Comment:
I couldn't find where the pendingDocIDs gets reset inside the Replicator class. It should get reset when the replicator changes its status. Here is where iOS does it.

https://github.com/couchbase/couchbase-lite-ios/blob/master/Source/API/CBLReplication.m#L349